libminifi/src/FlowController.cpp

```diff
@@ -558,19 +552,20 @@ std::vector<state::StateController*> FlowController::getAllProcessorControllers(
   return controllerVec;
 }
 
-state::StateController* FlowController::getProcessorController(const std::string& name, const std::function<std::unique_ptr<state::ProcessorController>(core::Processor&)>& controllerFactory) {
-  auto* processor = root_->findProcessorByName(name);
-  if (processor == nullptr) {
-    logger_->log_error("Could not get processor controller for requested name \"%s\", because processor was not found either", name);
-    return nullptr;
-  }
-
-  // reference to the existing or newly created controller
-  auto& foundController = processor_to_controller_[processor->getUUID()];
-  if (!foundController) {
-    foundController = controllerFactory(*processor);
-  }
-  return foundController.get();
+state::StateController* FlowController::getProcessorController(const std::string& id_or_name, const std::function<std::unique_ptr<state::ProcessorController>(core::Processor&)>& controllerFactory) {
```

Comment:
you added a `gsl::not_null` to the return type of the `controllerFactory` functor at other places, but not here; why?

Reply:
This was an oversight, fixed in [4d699a7](https://github.com/apache/nifi-minifi-cpp/pull/1388/commits/4d699a7c00834fe02e7775c6e362a5cc96447c62).